application/models/services/UploadValidator.php

```diff
@@ -0,0 +1,82 @@
+<?php
+
+namespace LimeSurvey\Models\Services;
+
+class UploadValidator
+{
+    /**
+     * Check uploaded file size
+     *
+     * @param string $fileName the name of the posted file
+     * @param mixed $customMaxSize maximum file upload size
+     *
+     * @return string|null the error message or null if all checks are ok
+     */
+    public function getError($fileName, $customMaxSize = null)
+    {
+        if (is_null($customMaxSize)) {
+            $maximumSize = getMaximumFileUploadSize();
+        } else {
+            $maximumSize = min((int) $customMaxSize, getMaximumFileUploadSize());
+        }
+
+        // When 'post_max_size' is exceeded $_POST and $_FILES are empty.
+        // There is no way to confirm if the superglobals are empty because 'post_max_size' was
+        // exceeded, or because nothing was posted.
+        if (empty($_POST) && empty($_FILES)) {
```

Comment:
Please inject $_POST and $_FILES instead of accessing them directly. E.g., you can add two attributes to the class, `private $post; private $files` and inject them in the constructor. This will make it possible to unit-test the function later.

Reply:
Check now please